docs/wiki/08a-examples-and-tests.md

```diff
@@ -0,0 +1,161 @@
+- [Configuring the Build Process](#configuring-the-build-process)
+  - [Including Examples and Unit Tests](#including-examples-and-unit-tests)
+  - [Excluding Examples and Unit Tests](#excluding-examples-and-unit-tests)
+- [Building the AREG SDK](#building-the-areg-sdk)
+- [Running AREG SDK Examples](#running-areg-sdk-examples)
+- [Running AREG SDK Unit Tests](#running-areg-sdk-unit-tests)
+- [Notes](#notes)
+
+---
+
+## Introduction
+
+By default, the AREG SDK build process includes Examples and Unit Tests to facilitate learning and testing.
+Developers can exclude these components to streamline the build process or focus solely on the SDK's core functionality.
+This document provides step-by-step instructions for configuring, building, and running the Examples and Unit Tests of the AREG SDK.
+
+---
+
+## Prerequisites
+
+Before proceeding, ensure the following requirements are met:
+
+- **AREG SDK Source Code**: Ensure the source code is available on your system.
+- **CMake**: Version 3.20 or newer.
+- **Java**: Version 17 or newer.
+- **C++ Compiler**: Supported options include GCC, Clang, or MSVC.
+- **Build Environment**: Set up a terminal or command prompt with the necessary tools.
+
+> [!NOTE]
+> The AREG SDK Examples and Unit Tests are included in the `areg-sdk.sln` Solution file of the Microsoft Visual Studio and are built automatically.
+
+---
+
+## Configuring the Build Process
+
+### Including Examples and Unit Tests
+
+To include **Examples** and **Unit Tests** in the build process, use the default configuration. No additional options are required:
+
+```bash
+cmake -B ./build
+cmake --build ./build
+```
+
+This configuration builds the AREG SDK core along with all Examples and Unit Tests.
+
+### Excluding Examples and Unit Tests
+
+To exclude specific components, use the following CMake options:
+
+- `AREG_BUILD_TESTS`: Controls the inclusion of Unit Tests.
+- `AREG_BUILD_EXAMPLES`: Controls the inclusion of Examples.
+
+Set these options to `OFF` to exclude them from the build process.
+
+**Example**: Exclude both Unit Tests and Examples:
+
+```bash
+cmake -B ./build -DAREG_BUILD_TESTS=OFF -DAREG_BUILD_EXAMPLES=OFF
+cmake --build ./build
+```
+
+---
+
+## Building the AREG SDK
+
+After configuring the build process, proceed as follows:
+
+1. **Generate Build Files**: Run the `cmake` command with appropriate options.
+   
+   **Example**: Build the SDK with Clang, excluding Unit Tests but including Examples:
+   ```bash
+   cmake -B ./build -DAREG_COMPILER_FAMILY=llvm -DAREG_BUILD_TESTS=OFF -DAREG_BUILD_EXAMPLES=ON
+   ```
+
+2. **Build the Project**: Execute the build command:
+   ```bash
+   cmake --build ./build
+   ```
+
+The binaries and artifacts will be available in the `./product/build/...` directory structure.
+
+---
+
+## Running AREG SDK Examples
+
+To execute Examples:
+
+1. Navigate to the `bin` directory inside `./product/build/...`.
+2. Locate the desired example binary.
+3. Run the binary from the terminal.
```

Comment:
I think the standard way to run the test is via cmake build system, like:
```
cd build/
make test
```
But then I am not sure how to pass the user options to the test program

Reply:
or (run with filter):
```
ctest --test-dir ./build -R DateTimeTest
``